\input{head.tex}
\usepackage[printonlyused,withpage,nohyperlinks]{acronym}

\begin{document}
\title{PanelAppRex aggregates disease gene panels and facilitates sophisticated search}	
\author[1]{Dylan Lawless\thanks{Addresses for correspondence: \href{mailto:Dylan.Lawless@kispi.uzh.ch}{Dylan.Lawless@kispi.uzh.ch}}}

\affil[1]{Department of Intensive Care and Neonatology, University Children's Hospital Zürich, University of Zürich, Switzerland.}

\maketitle
\justify

\maketitle
Word count: 1342

\begin{abstract}
\noindent
\textbf{Motivation:} Gene panel data provides critical insights into disease-gene correlations. However, aggregating and interrogating this diverse dataset can be challenging. PanelAppRex addresses this by first preparing a machine-readable aggregate and second by offering a sophisticated natural search interface that streamlines data exploration for both clinical and research applications.\\[1ex]
\textbf{Results:} PanelAppRex aggregates gene panel data from source including ClinVar, UniProt, and Genomics England’s PanelApp, including the approved panels used in the NHS National Genomic Test Directory and the 100,000 Genomes Project. It enables users to execute complex queries by gene names, phenotypes, disease groups and more, returning integrated datasets in multiple downloadable formats. Benchmarking demonstrates 
93\% - 100\% accuracy, effectively simplifying variant discovery and interpretation to enhance workflow efficiency. The greatest benefit is the analysis-ready format for bioinformatic integration. \\[1ex]
\noindent \textbf{Availability:} \url{https://switzerlandomics.ch/services/panelAppRexAi/} (A standalone webpage will be substituted for publication version).
The source code and data are accessible at \url{https://github.com/DylanLawless/PanelAppRex}. PanelAppRex is available under the MIT licence. 
The dataset is maintained for a minimum of two years following publication.
\end{abstract}
\clearpage


\section*{Acronyms}
\renewenvironment{description} % Internally acronym uses description which we redefine to make condense spacing. 
{\list{}{\labelwidth0pt\itemindent-\leftmargin
    \parsep-1em\itemsep0pt\let\makelabel\descriptionlabel}}
               {\endlist}
\begin{acronym} 
\acro{api}[API]{Application Programming Interface}
\acro{csv}[CSV]{comma-separated values}
\acro{ge}[GE]{Genomics England}
\acro{hgnc}[HGNC]{Human genome organisation Gene Nomenclature Committee}
\acro{html}[HTML]{HyperText Markup Language}
\acro{jaci}[JACI]{Journal of Allergy and Clinical Immunology}
\acro{mit}[MIT]{Massachusetts Institute of Technology}
\acro{moi}[MOI]{Mode of Inheritance}
\acro{nhs}[NHS]{National Health Service}
\acro{omim}[OMIM]{Online Mendelian Inheritance in Man}
\acro{pdf}[PDF]{Portable Document Format}
\acro{pid}[PID]{Primary immunodeficiency}
\acro{rag}[RAG]{Retrieval-augmented generation}
\acro{rds}[Rds]{R Data Serialization format}
\acro{tsv}[TSV]{Tab-separated values}
\end{acronym} 
 

\section{Introduction}
\noindent
Disease-gene panels are pivotal for the diagnosis and interpretation of genetic disorders. Sources like \ac{ge}’s PanelApp and PanelApp Australia host comprehensive panels that support genomic testing \cite{martin_panelapp_2019}. 
For instance, these are integral in the NHS and research projects such as the 100,000 Genomes Project\cite{martin_panelapp_2019}. Despite its utility, manual panel selection and data aggregation remain labour intensive. PanelAppRex was developed to simplify this process by automating data retrieval from \ac{api} sources and integrating the data into machine and user-friendly formats. We include the use of \ac{ge}’s PanelApp, ClinVar, and UniProt
\cite{martin_panelapp_2019,
landrum_clinvar_2018, the_uniprot_consortium_uniprot_2025}. 
The novel natural language-style search capability further streamlines the discovery of disease gene panels by allowing queries based on gene names, phenotypes, disease groups and additional key attributes which were supplemented with evidence-based \ac{rag}.


%\section{Introduction}
%Quantifying the risk that a newborn inherits a disease-causing variant is a fundamental challenge in genomics. Currently, genetic risk estimation is most commonly performed using classical statistical approaches grounded in Hardy–Weinberg equilibrium (HWE)
%\cite{MayoCentury2008, AbramovsHardyWeinberg2020}. 
%These methods are considered best practice for calculating genetic inheritance probabilities across the genome for single nucleotide variants (SNVs). Using \textit{TNFAIP3} as a model gene, we calculate the expected incidence of pathogenic variants in a cohort representative of the annual birth population in Switzerland.
%
%Our work lays the groundwork for future studies aimed at developing a Bayesian framework for variant occurrence. Such a framework would refine these classical estimates by incorporating prior knowledge and continuously updating probabilities as new data become available. Moreover, our approach builds on several key aspects of current variant interpretation and statistical genomics. For example, guidelines from the American College of Medical Genetics and Genomics (ACMG) \citep{richards2015standards, tavtigian2020fitting} provide a foundation for classifying variants into categories (Pathogenic, Likely Pathogenic, VUS, Likely Benign, and Benign). Standardised variant interpretation protocols further integrate quality control and filtering criteria to ensure robust genomic analysis \citep{pedersen2021effective,anderson2010data}. In parallel, the concept of qualifying variants (QVs) is important for suitably filtering and interpreting SNVs \citep{cirulli2015exome, Povysil2019rare}. These components, together with advanced statistical approaches (e.g., ACAT and SKAT \citep{liu2019acat,li2020dynamic,wu2011rare,lee2012optimal}) and multi-block data fusion techniques \citep{kong2018nature,howe2021within}, form the pillars of current practices in variant interpretation. Furthermore, standardized reporting formats and unique identifiers for QV sets (e.g., using ACMG SF v3.2 \citep{miller2023acmg}) enhance reproducibility and interoperability. These concepts underpin our methodology and provide a strong foundation for future Bayesian extensions.







%\section{Introduction2}
%
%Advances in genomic sequencing have enabled the comprehensive characterisation of genetic variation across populations, yet the clinical interpretation of these variants remains challenging. While true positive pathogenic variants are generally straightforward to confirm, benign variants or variants of uncertain significance (VUS) pose a greater challenge. These variants may be mistakenly given undue importance simply because they occur in genes known to be associated with disease, even if their occurrence in the general population is merely by chance.
%
%In this study, we integrate a Bayesian framework with classical probability models to refine the estimation of disease occurrence. By explicitly modelling the empirical occurrence probability of every variant-including benign and VUS variants-we ensure that the expected prior probability is appropriately incorporated into pathogenicity assessments. This approach provides a more balanced view that aids clinical interpretation by preventing overestimation of the clinical relevance of common variants.
%
%We demonstrate our integrated framework using examples from autosomal dominant (\textit{NFKB1}) and autosomal recessive (CFTR p.Arg117His) disorders. This framework not only improves our understanding of the true burden of disease but also offers clinical practitioners a cautious perspective, particularly when interpreting variants that might otherwise be erroneously prioritized.


\section{Materials and methods}

\subsection{Data}
The PanelAppRex core model contained 58,592 entries consisting of 52 sets of annotations, including the gene name, disease-gene panel ID, diseases-related features, confidence measurements \cite{lawless_panelapprex_2025}.
Data from gnomAD v4 comprised 807,162 individuals, including 730,947 exomes and 76,215 genomes \cite{karczewski2020mutational}. This dataset provided 786,500,648 single nucleotide variants and 122,583,462 InDels, with variant type counts of 9,643,254 synonymous, 16,412,219 missense, 726,924 nonsense, 1,186,588 frameshift and 542,514 canonical splice site variants. ClinVar data were obtained from the variant summary dataset (this version: 16 March 2025) available from the NCBI FTP site, and included 6,845,091 entries, which were processed into 91,319 gene classification groups and a total of 38,983 gene classifications; for example, the gene \textit{A1BG} contained four variants classified as likely benign and 102 total entries \cite{landrum_clinvar_2018}. For our analysis phase we also used dbNSFP which consisted of a number of annotations for 121,832,908 single nucleotide variants 
\cite{liu_dbnsfp_2020}. 

\subsection{Implementation}
\noindent
PanelAppRex was implemented in R and integrates data from \ac{ge}’s PanelApp, ClinVar, and UniProt
\cite{martin_panelapp_2019,
landrum_clinvar_2018, the_uniprot_consortium_uniprot_2025}. 
It performed credentialed access to the API to retrieve all approved panels, merging them into two formats: a simplified version (Panel ID, Gene) and a complex version (including metadata such as confidence level, mode of inheritance, and disease information), and several metadata summary statistics. In addition, the tool incorporates a search module to execute complex user queries. The search functionality supports queries by gene names, phenotypes, disease names, disease groups, panel names, genomic locations and other identifiers. 
\ac{rag} was used to improve the natural queries in hidden states based on evidence about the disease and gene function by supplementing the data with additional sources including ClinVar, UniProt, etc. 

\subsection{Usage}
\noindent
On a desktop browser, queries can be executed via the integrated search bar in our \ac{html} version, where a JavaScript function splits the query into individual terms and progressively filters rows - retaining only those that match all active terms while ignoring unmatched ones. This enables users to perform complex, partial matching queries (e.g. ``paediatric \textit{RAG1} primary immunodeficiency skin disorder'') to rapidly identify the panel most closely associated with their hypothesis on primary immunodeficiency and paediatric skin disorders.

Bioinformatically, users can import the provided, ready-for-use, datasets in TSV or Rds formats. 
A typical use case might involve users merging with their own omic data based on gene or protein ID.
The following code snippet, available in \texttt{minimal\_example.R}, demonstrates how to load the data in R:
\begin{verbatim}
# TSV format
path_data <- "../data"
core_path <- paste0(path_data, "/PanelAppData_combined_core")
minimal_path <- paste0(path_data, "/PanelAppData_combined_minimal")

df_core <- read.table(
    file= paste0(core_path, ".tsv"), 
    sep = "\t", header = TRUE)

df_minimal <- read.table(
    file= paste0(minimal_path, ".tsv"), 
    sep = "\t", header = TRUE)

# Rds format
rds_path <- paste0(path_data, "/PanelAppData_combined_Rds")
df_core <- readRDS(file= rds_path)
\end{verbatim}

\subsection{Validation for completeness in core data}
\noindent
To ensure the reliability and completeness of the core dataset, we systematically assessed whether key gene-level fields were present for each entry after merging the core dataset.
Specifically, we checked for non-missing values in gene symbols, associated publications, disease panel names, \ac{moi}, and \ac{omim} gene IDs. 
Where \ac{hgnc} or Ensembl gene IDs were missing, we used programmatic queries to the Ensembl database via the \texttt{biomaRt} R package to recover the identifiers using the available \ac{hgnc} symbol as input \cite{steffen_durinck_biomart}. 
This validation step was applied to the full integrated PanelAppRex dataset.

\subsection{Benchmarking for manual queries}
\noindent
To mimic a clinician diagnosing a new disease such as \ac{pid}, we began by systematically selecting genetic diagnosis case studies from the current online catalogue from the \ac{jaci}, using the first five results  \cite{arruda_genetic_2015, 
mcaleer_severe_2015,
verhoeven_hematopoietic_2022,
magerus-chatinet_autoimmune_2013,
sharfe_fatal_2014}. 
Although the method presented here is generalisable to genetic diagnosis across disease areas, we chose to validate it using \ac{pid} case studies, as this is the primary focus of our own research.

The clinical background from these studies was used to construct keyword queries from patient features, simulating a naïve starting position for a clinician.
We then tested whether our PanelAppRex tool could successfully retrieve panels that included the final causal gene reported in each case study. 
The sources, queries, and results are shown in \textbf{Table \ref{tab:benchmark}}. 

\section{Results}
\noindent
\subsection{Core dataset}
PanelAppRex successfully aggregated data, currently from 451 panels, and several genomics databases to offer a user-friendly search functionality (\textbf{Figure \ref{fig:performance}}).
Users can retrieve results filtered by gene names, phenotypes, disease groups and other criteria. 
The system returned a table view with panel details and provided options for exporting results in \ac{csv}, Excel, or \ac{pdf} formats.
Bioinformatic uses may include generating virtual panels, constructing prior odds, or supporting formal reporting in qualifying variant protocols.

\begin{figure}[ht]
    \centering
    \includegraphics[width=0.49\textwidth]{screenshot_1.png}
    \includegraphics[width=0.49\textwidth]{screenshot_2.png}    
    \includegraphics[width=0.85\textwidth]{benchmark.pdf}    
\caption{\textbf{PanelAppRex interface displaying search results for a complex query.} Top: screenshots showing the search interface, with the left panel displaying the full database before filtering and the right panel showing detailed results for a selected panel. Bottom: benchmark metrics are presented as follows: (A) for each of the 5 case studies, the total number of panels returned, the number of panels that included the causal gene, and the single best panel selected (always 1 by default); (B) the percentage of all returned panels that included the true causal gene; (C) the percentage of the single best panels that contained the true causal gene.}
    \label{fig:performance}
\end{figure}

% \FloatBarrier
\subsection{Validation shows completeness in the core dataset}
\noindent
The raw dataset exhibited near-complete coverage in most fields after merging the core dataset. 
Specifically, 99.9915\% of gene entries had \ac{hgnc} symbols % (58587/58592)*100
and 99.2315\% had Ensembl gene IDs. % (57844/58292)*100
All entries included at least one publication, a disease panel name, a mode of inheritance, and an \ac{omim} gene ID (100\% each). 
After recovery of missing identifiers, completeness reached 100\% across all these core fields.
This validated dataset ensures that users can confidently build on a consistent foundation, using standardised and stable identifiers to link or enrich the core data with external resources (\textbf{Figure \ref{fig:validation}}).

\begin{figure}[ht]
    \centering
    \includegraphics[width=0.99\textwidth]{validation_counts.pdf}
\caption{\textbf{Validation and recovery of core annotation fields in the PanelAppRex dataset.}
(A) Unique genes and HGNC IDs before and after retrieving missing HGNC entries via Ensembl.
(B) Availability of publication annotations for genes.
(C) Unique genes and Ensembl gene IDs before and after biomart-based recovery of missing Ensembl IDs.
(D) Gene entries with associated disease panel names.
(E) Gene entries with annotated mode of inheritance.
(F) Gene entries linked to an OMIM gene ID.
Each bar shows the count of entries with non-missing values for the respective field. Updated fields (HGNC and Ensembl ID) reflect values recovered via external lookup using HGNC symbols.
}
    \label{fig:validation}
\end{figure}

\subsection{Benchmarking for manual queries confirms accurate retrieval of causal gene}
\noindent
To evaluate the practical utility of our system on a desktop browser, we applied the benchmarking approach described in methods using five published case studies of \ac{pid}
\cite{arruda_genetic_2015, 
mcaleer_severe_2015,
verhoeven_hematopoietic_2022,
magerus-chatinet_autoimmune_2013,
sharfe_fatal_2014}. % results
The tool's query process retrieved gene panels using natural language-style input. 
For example, in the first case study on Hereditary Angioedema, the authors reported suspecting that the condition was linked to ``\textit{SERPING1}, Factor XII, and edema" which we used as the query terms. 
Although the true causal gene term \textit{F12} (using its official \ac{hgnc} name) was not mentioned, the inclusion of the related disease terms enabled the system to correctly retrieve the panel containing \textit{F12} based on the hidden search knowledge.

While an expert might readily recognise ``Factor XII'' as referring to \textit{F12}, this correct query result demonstrates that the system can make such connections automatically, supporting users with varying levels of genetic expertise.

In our evaluation, PanelAppRex returned panels in which the causal gene was present in 93\% of all returned panels, with the subjective user-selected ``best panel'' achieving a perfect accuracy of 100\% 
(\textbf{Figure \ref{fig:performance}}).
 These metrics were derived by comparing the causal gene identified from the case study abstracts to the panels returned by our query, and by applying a subjective relevance measure to mimic intuitive panel selection - acknowledging that certain panels, such as the established PID gene panel, are inherently more reliable than broader, less specific panels. Overall, the results confirm that our approach accurately identifies the most relevant panels and effectively supports clinical decision-making in complex diagnostic scenarios.

\begin{table}[htbp]
\caption{\textbf{Summary of case study queries and PanelAppRex results.} *Case study 5 had five individual cases and patient information was significantly longer that other studies. It was therefore converted to keyword query automatically by the OpenAI model o3-mini to remove subjective bias and align with the other queries.}
\label{tab:benchmark}
\centering
\resizebox{\textwidth}{!}{
\begin{tabular}{|
    p{1cm}|
    p{4cm}|
    p{4cm}|
    p{1.4cm}|
    p{1.2cm}|
    p{1.2cm}|
    p{1.2cm}|
    p{1.2cm}|
    p{1.2cm}|
    p{5cm}|}
\hline
\textbf{Case study}	(\textbf{Ref})	 & 	\textbf{Source title}	 & 	\textbf{Query}	 & 	\textbf{Causal gene}	 & 	\textbf{Result panels}	 & 	\textbf{Subjective best panels}	 & 	\textbf{Panels with causal gene}	 & 	\textbf{Subjective relevance ratio}	 & 	\textbf{Causal gene in all results}	 & 	\textbf{Result ID, panelName, geneCount} \\
\hline																			
1	\cite{arruda_genetic_2015}	 & 	Genetic Analysis As a Practical Tool to Diagnose Hereditary Angioedema with Normal C1 Inhibitor: A Case Report	 & 	SERPING1 Factor XII edema	 & 	F12	 & 	3	 & 	1	 & 	3	 & 	0.3	 & 	1	 & 	64 COVID-19 research 695; 192 Primary immunodeficiency or monogenic inflammatory bowel disease 572; 311 Research panel - Severe Paediatric Disorders 2691 \\
\hline																			
2	\cite{mcaleer_severe_2015}	 & 	Severe dermatitis, multiple allergies, and metabolic wasting syndrome caused by a novel mutation in the N-terminal plakin domain of desmoplakin	 & 	SAM syndrome DSG1 dermatitis metabolic wasting	 & 	DSP	 & 	3	 & 	3	 & 	3	 & 	1	 & 	1	 & 	205 Fetal anomalies 2185; 210 DDG2P 2422; 211 Paediatric disorders 3903 \\
\hline																			
3	\cite{verhoeven_hematopoietic_2022}	 & 	Hematopoietic stem cell transplantation in a patient with proteasome-associated autoinflammatory syndrome (PRAAS)	 & 	resistant cutaneous vasculitis SH2D1A	 & 	PSMB4	 & 	3	 & 	1	 & 	2	 & 	0.3	 & 	0.7	 & 	64 COVID-19 research 695; 192 Primary immunodeficiency or monogenic inflammatory bowel disease 572; 311 Research panel - Severe Paediatric Disorders 2691 \\
\hline																			
4	\cite{magerus-chatinet_autoimmune_2013}	 & 	Autoimmune lymphoproliferative syndrome caused by a homozygous null FAS ligand (FASLG) mutation	 & 	Autoimmune lymphoproliferative syndrome ALPS lymphoproliferation hypergammaglobulinemia autoimmune cytopenia	 & 	FASLG	 & 	2	 & 	1	 & 	2	 & 	0.5	 & 	1	 & 	64 COVID-19 research 695; 192 Primary immunodeficiency or monogenic inflammatory bowel disease 572 \\
\hline																			
5*	\cite{sharfe_fatal_2014}	 & 	Fatal combined immunodeficiency associated with heterozygous mutation in STAT1	 & 	primary immunodeficiency recurrent pneumonia chronic diarrhea oral thrush bronchiectasis lymphadenopathy hepatosplenomegaly autoimmune hepatitis Addison	 & 	STAT1	 & 	1	 & 	1	 & 	1	 & 	1	 & 	1	 & 	192 Primary immunodeficiency or monogenic inflammatory bowel disease 572 \\
\hline
\end{tabular}
}
\end{table}

\FloatBarrier
\section{Discussion and conclusion}
\noindent
PanelAppRex provides an accessible and validated platform for querying, aggregating, and exporting curated disease-gene panel data. Designed for both clinical and research use, it simplifies study planning and variant interpretation through a user-friendly interface, while also offering a harmonised dataset for programmatic workflows. The core model integrates over 58,000 panel-gene associations, with full coverage of key fields including \ac{hgnc} symbols, Ensembl IDs, \ac{omim} annotations, \ac{moi}, and supporting publications.

Beyond immediate usability, PanelAppRex lays a foundation for genome-wide statistical modelling of disease risk. 
With annotations for gene-disease relationships, the dataset can support rigorous estimation of the prior probability of observing  variant classifications (e.g. benign, pathogenic) under different \ac{moi}.
This helps address a persistent challenge in clinical genetics: the lack of principled priors for variant interpretation that account not only for known pathogenic variants (true positives), but also for unobserved variants and negative evidence (false negative causal pathogenic variant and true negative absence of causal variants) \cite{hannah_using_2024, zschocke_mendelian_2023}. 
By integrating high-resolution allele frequencies from gnomAD \cite{karczewski2020mutational}, curated classifications from ClinVar \cite{landrum_clinvar_2018}, and structured gene-disease associations \cite{martin_panelapp_2019}, 
PanelAppRex can enable the derivation of calibrated genome-wide priors for Bayesian models of genetic risk.

These capabilities go beyond panel lookup to support evidence-aware diagnostics. 
Probabilistic models incorporating both observed and unobserved variation can quantify uncertainty, resolve ambiguous findings, and refine variant prioritisation, scaling naturally to support complex, multi-gene disorders. 
The dataset also provides a substrate for AI-driven approaches to variant interpretation, including probabilistic inference, reinforcement learning, and deep annotation pipelines
\cite{jumper_highly_2021, cheng_accurate_2023}.

Several limitations should be acknowledged. Not all known coding genes are currently linked to a disease panel; we prioritised high-confidence, traceable annotations over broad but less reliable coverage. 
Some genes are over-represented across multiple panels due to historical research biases, leading to non-uniform panel enrichment. 
Additionally, not all panels returned by queries may be equally informative. 
For example, during benchmarking, the COVID-19 research panel frequently appeared alongside the \ac{pid} panel due to overlapping genes. 
While technically accurate, such panels may be less relevant to users focused on clinical  diagnosis of immunodeficiency.
% To consider typical user expectations, we included a subjective ``best match'' choice in each case study.
In bioinformatic workflows, these choices can be refined systematically or excluded using filters or downstream logic.

PanelAppRex bridges expert-curated panel knowledge with genome-scale statistical reasoning. 
It offers a robust tool for interactive search, a validated dataset for programmatic access, and a scalable framework for quantitative genomic modelling. 
Future work will focus on expanding supported queries, integrating additional variant types and annotations, and enabling more sophisticated applications in variant interpretation and risk estimation.

\section{Conclusion}
\noindent
PanelAppRex offers a robust solution for aggregating and querying gene panel data. Its sophisticated search feature simplifies data exploration and enhances variant interpretation. 

\section*{Acknowledgements}
\noindent
We acknowledge \ac{ge} for providing public access to the PanelApp data.
The use of data from \ac{ge} panelapp was licensed under the Apache License 2.0.
The use of data from UniProt was licensed under Creative Commons Attribution 4.0 International (CC BY 4.0).
ClinVar asks its users who distribute or copy data to provide attribution to them as a data source in publications and websites \cite{landrum_clinvar_2018}.

\section*{Competing interest}
\noindent
We declare no competing interest. 

\bibliographystyle{unsrtnat}
\bibliography{references.bib}
\clearpage

\end{document}