\input{head.tex}
\usepackage[printonlyused,withpage,nohyperlinks]{acronym}

\begin{document}
\title{Supplemental - PanelAppRex aggregates disease gene panels and facilitates sophisticated search}	

% in your preamble, before \author:
\newcommand{\QUANT}{1}
\newcommand{\GHI}{2}
\newcommand{\LEEDS}{3}
\newcommand{\IPSNEO}{4}

\author[\QUANT]{Quant Group} 
\author[\GHI]{Simon Boutry}% \textsuperscript{†}}
\author[\GHI]{Ali Saadat}% \textsuperscript{†}} \textsuperscript{†}}
\author[\LEEDS]{Sinisa Savic}% orcid.org/0000-0001-7910-0554
\author[\IPSNEO]{Luregn J. Schlapbach}
\author[\GHI]{Jacques Fellay}
\author[\IPSNEO]{Dylan Lawless \thanks{Addresses for correspondence: \href{mailto:Dylan.Lawless@kispi.uzh.ch}{Dylan.Lawless@kispi.uzh.ch}}
}

%\author[\KISPIIMM]{Maarja Soomann}% \textsuperscript{†}} \textsuperscript{†}}
%\author[\KISPIIMM]{Johannes Trück}
\affil[\QUANT]{The quantitative omic epidemiology group.} 
\affil[\GHI]{Global Health Institute, School of Life Sciences, École Polytechnique Fédérale de Lausanne, Switzerland.}
\affil[\LEEDS]{Leeds Institute of Rheumatic and Musculoskeletal Medicine, University of Leeds, Leeds, UK.}
\affil[\IPSNEO]{Department of Intensive Care and Neonatology, University Children's Hospital Zurich, University of Zurich, Zurich, Switzerland.}

\maketitle
\justify

%\\\\\\\\\\\\\\\\\\\\\\\\\\\\
\clearpage
\beginsupplement
\section*{Supplemental} \label{Supplemental_text}


\begin{table}[htbp]
\caption{\textbf{Summary of case study queries and PanelAppRex results.} 
``Subjective best panels'' are those reasonably preferable to the clinical query and unlikely to be excluded by users. In the benchmark scenario, broad or less relevant panels (e.g. ``COVID-19 research'') might be deprioritised in favour of more clinically aligned options such as ``Fetal anomalies'', or ``Paediatric disorders''. Summarised in \textbf{Figure 1}.
*Case study 5 had five individual cases and patient information was significantly longer than other studies. Therefore, we used OpenAI model o3-mini to converted it to a standardised keyword query automatically, thereby removing our subjective bias and aligning with the other queries.
}
\label{tab:benchmark}
\centering
\resizebox{\textwidth}{!}{
\begin{tabular}{|
    p{1cm}|
    p{4cm}|
    p{4cm}|
    p{1.4cm}|
    p{1.3cm}|
    p{1.3cm}|
    p{1.3cm}|
    p{1.3cm}|
    p{1.3cm}|
    p{5cm}|}
\hline
\textbf{Case study}	(\textbf{Ref})	 & 	\textbf{Source title}	 & 	\textbf{Query}	 & 	\textbf{Causal gene}	 & 	\textbf{Result panels}	 & 	\textbf{Subjective best panels}	 & 	\textbf{Panels with causal gene}	 & 	\textbf{Subjective relevance ratio}	 & 	\textbf{Causal gene in all results}	 & 	\textbf{Result ID, panelName, geneCount} \\
\hline																			
1	\cite{arruda_genetic_2015}	 & 	Genetic Analysis As a Practical Tool to Diagnose Hereditary Angioedema with Normal C1 Inhibitor: A Case Report	 & 	SERPING1 Factor XII edema	 & 	F12	 & 	3	 & 	1	 & 	3	 & 	0.3	 & 	1	 & 	64 COVID-19 research 695; 192 Primary immunodeficiency or monogenic inflammatory bowel disease 572; 311 Research panel - Severe Paediatric Disorders 2691 \\
\hline																			
2	\cite{mcaleer_severe_2015}	 & 	Severe dermatitis, multiple allergies, and metabolic wasting syndrome caused by a novel mutation in the N-terminal plakin domain of desmoplakin	 & 	SAM syndrome DSG1 dermatitis metabolic wasting	 & 	DSP	 & 	3	 & 	3	 & 	3	 & 	1	 & 	1	 & 	205 Fetal anomalies 2185; 210 DDG2P 2422; 211 Paediatric disorders 3903 \\
\hline																			
3	\cite{verhoeven_hematopoietic_2022}	 & 	Hematopoietic stem cell transplantation in a patient with proteasome-associated autoinflammatory syndrome (PRAAS)	 & 	resistant cutaneous vasculitis SH2D1A	 & 	PSMB4	 & 	3	 & 	1	 & 	2	 & 	0.3	 & 	0.7	 & 	64 COVID-19 research 695; 192 Primary immunodeficiency or monogenic inflammatory bowel disease 572; 311 Research panel - Severe Paediatric Disorders 2691 \\
\hline																			
4	\cite{magerus-chatinet_autoimmune_2013}	 & 	Autoimmune lymphoproliferative syndrome caused by a homozygous null FAS ligand (FASLG) mutation	 & 	Autoimmune lymphoproliferative syndrome ALPS lymphoproliferation hypergammaglobulinemia autoimmune cytopenia	 & 	FASLG	 & 	2	 & 	1	 & 	2	 & 	0.5	 & 	1	 & 	64 COVID-19 research 695; 192 Primary immunodeficiency or monogenic inflammatory bowel disease 572 \\
\hline																			
5*	\cite{sharfe_fatal_2014}	 & 	Fatal combined immunodeficiency associated with heterozygous mutation in STAT1	 & 	primary immunodeficiency recurrent pneumonia chronic diarrhea oral thrush bronchiectasis lymphadenopathy hepatosplenomegaly autoimmune hepatitis Addison	 & 	STAT1	 & 	1	 & 	1	 & 	1	 & 	1	 & 	1	 & 	192 Primary immunodeficiency or monogenic inflammatory bowel disease 572 \\
\hline
\end{tabular}
}
\end{table}





\begin{figure}[ht]
    \centering
    \includegraphics[width=0.99\textwidth]{validation_counts.pdf}
\caption{\textbf{Validation and recovery of core annotation fields in the PanelAppRex dataset.}
(A) Unique genes and HGNC IDs before and after retrieving missing HGNC entries via Ensembl.
(B) Availability of publication annotations for genes.
(C) Unique genes and Ensembl gene IDs before and after biomart-based recovery of missing Ensembl IDs.
(D) Gene entries with associated disease panel names.
(E) Gene entries with annotated \ac{moi}.
(F) Gene entries linked to an OMIM gene ID.
Each bar shows the count of entries with non-missing values for the respective field. Updated fields (HGNC and Ensembl ID) reflect values recovered via external lookup using HGNC symbols. Percentage shows the complete recovery of coverage across features.
}
    \label{fig:validation}
\end{figure}




















\begin{figure}[ht]
    \centering
    \includegraphics[width=0.75\textwidth]{plot_patch2_annotated_example.png}    
\caption{\textbf{Summary of gene confidence levels, reuse across panels, and panel sizes in the PanelAppRex dataset.}
(A) Number of genes per confidence level. 
(B) Number of panels in which each gene is included, with the example gene \textit{RAG1} highlighted to demonstrate that it is present in 7 panels. 
(C) Number of genes per panel, with two representative panels annotated: panel ID 398 (\ac{pid}/\ac{iei}) and panel ID 1220 (\ac{iem}).
}
    \label{fig:summary_stats}
\end{figure}


\begin{figure}[ht]
    \centering
    \includegraphics[width=0.99\textwidth]{plot_patch_uniq_gene_moi_summary.pdf}
\caption{\textbf{Summary of mode of inheritance annotations in the PanelAppRex dataset.}
(A) Counts of unique genes annotated with each \ac{moi}, based on non-redundant gene-\ac{moi} combinations.
(B) Total number of unique genes and total number of gene-\ac{moi} combinations in the harmonised PanelAppRex dataset.
}
    \label{fig:uniq_gene_moi}
\end{figure}

\begin{figure}[ht]
    \centering
    \includegraphics[width=0.99\textwidth]{panelapprex_logo_v2_16x9.png}
\caption{\textbf{(logo)} PanelAppRex reigns over genomic complexity.
}
    \label{fig:uniq_gene_moi}
\end{figure}

\FloatBarrier
\clearpage
\bibliographystyle{unsrtnat}
\bibliography{references.bib}


\end{document}
