\documentclass[12pt,a4paper]{letter}
\usepackage{graphicx}
\usepackage[utf8]{inputenc}
\usepackage{microtype}
\usepackage{amsmath, amsfonts, amssymb}
\usepackage[backend=bibtex, style=authoryear, natbib=true, sorting=nyt]{biblatex} 
\usepackage[colorlinks=true]{hyperref}
\hypersetup{
    linkcolor=blue,         % color of internal links
    citecolor=black,         % color of citation links
    urlcolor=blue           % color of URL links
}
\usepackage{filecontents}
\usepackage{geometry}
\geometry{left=3cm, top=2cm, right=3cm, bottom=2cm} 

\address{Dylan Lawless@kispi.uzh.ch\\
Department of Intensive\\
Care and Neonatology,\\
University Children's\\
Hospital Zürich,\\
University of Zürich.
}

\signature{Dylan Lawless, PhD\\
on behalf of all co-authors} 

\begin{document} 

\begin{letter}{Dear Editors,}

\opening{}

Accurate and systematic analysis of genomic data often relies on curated disease gene panels, yet these remain \textbf{fragmented} and \textbf{difficult to integrate} into clinical workflows. Our manuscript, \textit{``PanelAppRex aggregates disease gene panels and facilitates sophisticated search''}, introduces the first \textbf{openly available} harmonised platform of over \textbf{58,000 expert-curated} gene–disease associations, including all NHS-approved panels, consolidated into a \textbf{machine learning-ready} dataset with a \textbf{natural language search} interface.

\begin{center}
\includegraphics[width=\linewidth]{../images/panelapprex_logo_v2_16x9.png}
\noindent
\textbf{Figure:}\textit{ (logo) PanelAppRex reigns over genomic complexity.}
\bigskip
\end{center}

Our results demonstrate validation and benchmarking with:
\begin{itemize}
\item \textbf{Standardised} gene identifiers (HGNC, Ensembl, OMIM).
\item Structured \textbf{annotations} for inheritance, disease terms, and literature.
\item \textbf{Machine-readable} exports in TSV, CSV, and R.
\item Interactive, intuitive \textbf{search} and \textbf{filtering}.
\item Fully \textbf{open-source} and \textbf{freely accessible} for immediate clinical and research use.
\item Import-ready \textbf{complete database}.
\end{itemize}

PanelAppRex serves both technical and clinical users. Bioinformaticians can integrate it into variant prioritisation, modelling, and machine learning workflows. Clinicians and laboratory scientists can rapidly identify gene panels from partial clinical descriptions. In tests on published inborn errors of immunity cases, PanelAppRex retrieved the correct panels in all scenarios, even when gene names were absent.

We believe it suits an \textit{Application Note} in \textit{Bioinformatics}, contributing to precision medicine and genomic interpretation for both academic and industrial researchers

The manuscript is original, approved by all authors, and not under consideration elsewhere. Thank you for your consideration.

\closing{Yours sincerely,}

\end{letter}
\end{document}

%\cc{Cclist} 
%\ps{adding a postscript} 
%\encl{list of enclosed material} 
%\vfill
%\printbibliography[heading=none]